### PR TITLE
Fixup GMB location ID to parse uint64 IDs

### PR DESCRIPTION
**Pull request by nwbirnie**
See tg/1258621

This is pretty simple to do in java - just need to parse from string since it's an unsafe operation.

### CODE REVIEW

**Review by jradcliff — APPROVED** [review]

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddSmartCampaign.java [hunk 1]
```diff
@@ -185,17 +185,24 @@ private void runExample(
...
       String keyword,
       String freeFormKeywordText,
-      Long locationId,
+      String locationId,
       String businessName) {
     if (locationId == null && businessName == null) {
       throw new RuntimeException("Must provider either --locationId or --businessName");
     }
+
+    // [START smart_campaign_14]
```
Comment (BenRKarl) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddSmartCampaign.java:194]: Can you change this to `add_smart_campaign_14`

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddSmartCampaign.java [hunk 2]
```diff
@@ -185,17 +185,24 @@ private void runExample(
...
+    // Converts the location ID to unsigned long (potentially contains IDs with the most significant
+    // bit set).
+    long locationIdParsed = Long.parseUnsignedLong(locationId);
+    // [END smart_campaign_14]
```
Comment (BenRKarl) [inline google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddSmartCampaign.java:198]: This should be `add_smart_campaign_14`

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddSmartCampaign.java (hunk 1 above)
Comment (nwbirnie) [reply google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddSmartCampaign.java:194]: Done

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddSmartCampaign.java (hunk 2 above)
Comment (nwbirnie) [reply google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddSmartCampaign.java:198]: Done

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddSmartCampaign.java [hunk 3]
```diff
@@ -185,17 +185,24 @@ private void runExample(
...
+      String locationId,
       String businessName) {
     if (locationId == null && businessName == null) {
       throw new RuntimeException("Must provider either --locationId or --businessName");
     }
+
+    // [START smart_campaign_14]
+    // Converts the location ID to unsigned long (potentially contains IDs with the most significant
+    // bit set).
+    long locationIdParsed = Long.parseUnsignedLong(locationId);
```
[the remark this answers is not in the document]
Comment (nwbirnie) [reply google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddSmartCampaign.java:197]: This is correct then, it's parsing an unsigned long as a signed long.

**Review by BenRKarl — APPROVED** [review]

File: google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddSmartCampaign.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (wfansh) [reply google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddSmartCampaign.java:197]: This line throws **NumberFormatException** if `locationId` param is not provided (using `businessName` instead)

**Pull request merged**